File: 1395-count-number-of-teams/1395-count-number-of-teams.py
class Solution(object):
    def numTeams(self, rating):
        """
        :type rating: List[int]
        :rtype: int
        """
        total=0
        n=len(rating)
        for i in range(0,n):
             lm=0
             ll=0
             rm=0
             rl=0
             for j in range(i+1,n):
                if rating[j]<rating[i]:
                    lm+=1
                elif rating[j]>rating[i]:
                    ll+=1
             for j in range(i):
                if rating[j]<rating[i]:
                    rm+=1
                elif rating[j]>rating[i]:
                    rl+=1
             total+=lm*rl+ll*rm
        return total
            

File: 1395-count-number-of-teams/1395-count-number-of-teams.cpp
class Solution {
public:
    int numTeams(vector<int>& rating) {
        int n=rating.size();
        int total=0;
    for(int i=0;i<n;i++)
    {
        int lmost=0,lleast=0,rmost=0,rleast=0;
        for(int j=i+1;j<n;j++)
        {
            if(rating[j]<rating[i]) lmost++;
            else if(rating[j]>rating[i]) lleast++;
        }
         for(int j=0;j<i;j++)
        {
            if(rating[j]<rating[i]) rmost++;
            else if(rating[j]>rating[i]) rleast++;
        }
        total+=lmost*rleast+lleast*rmost;
    }
        return total;
    }
};